src/rayTracing.py

```diff
@@ -0,0 +1,127 @@
+# /*************************************************************************
+#  * Copyright (C) 2012-2014 Tavian Barnes <tavianator@tavianator.com>     *
+#  * https://github.com/tavianator/dimension                               *
+#  * Copyright (C) 2020 Pero Prebeg <Pero.Prebeg@fsb.hr>                   *
+#  *                                                                       *
+#  * The content of this file is part of The Dimension Library. The        *
+#  * original C Code of The Dimensions Library has been translated to      *
+#  * Python code, further accomodations have been made to emphasize the    *
+#  * purpose of d3v.                                                       *
+#  *                                                                       *
+#  * The Dimension Library is free software; you can redistribute it and/  *
+#  * or modify it under the terms of the GNU Lesser General Public License *
+#  * as published by the Free Software Foundation; either version 3 of the *
+#  * License, or (at your option) any later version.                       *
+#  *                                                                       *
+#  * The Dimension Library is distributed in the hope that it will be      *
+#  * useful, but WITHOUT ANY WARRANTY; without even the implied warranty   *
+#  * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU  *
+#  * Lesser General Public License for more details.                       *
+#  *                                                                       *
+#  * You should have received a copy of the GNU Lesser General Public      *
+#  * License along with this program.  If not, see                         *
+#  * <http://www.gnu.org/licenses/>.                                       *
+#  *************************************************************************/
+
+import math
+
+from bounds import BBox
+from pyVector3 import pyvector3
+
+
+class dmnsn_ray:
+    def __init__(self, los):
+        self._x0 = pyvector3()
+        self._x0.setFromQt(los[0])  # P0
+        self._n = pyvector3()
+        self._n.setFromQt(los[1])  # K
+
+    @property
+    def x0(self):
+        return self._x0
+
+    @property
+    def n(self):
+        return self._n
+
+
+class dmnsn_optimized_ray:
+    def __init__(self, ray: dmnsn_ray):
+        self._x0 = ray.x0
+        self._n_inv = pyvector3()
+        self._n_inv.setFromScalars((1.0 / ray.n.X) if ray.n.X != 0 else math.inf,
+                                   (1.0 / ray.n.Y) if ray.n.X != 0 else math.inf,
+                                   (1.0 / ray.n.Z) if ray.n.Z != 0 else math.inf)
+
+    @property
+    def x0(self):
+        return self._x0
+
+    @property
+    def n_inv(self):
+        return self._n_inv
+
+
+class dmnsn_aabb:
```

Comment:
yet again, what was the rationale behind designing new class, instead of using BBox?
It has a similar (but stripped down) API, and couple of new member functions. Have you considered deriving it from BBox ?

Reply:
Done, replaced by a class with clearer name and derivation from BBox